src/internal/operations/pathtransformer/restore_path_transformer.go

```diff
@@ -0,0 +1,180 @@
+package pathtransformer
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/internal/version"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+func locationRef(
+	ent *details.Entry,
+	repoRef path.Path,
+	backupVersion int,
+) (*path.Builder, error) {
+	loc := ent.LocationRef
+
+	// At this backup version all data types should populate LocationRef.
+	if len(loc) > 0 || backupVersion >= version.OneDrive7LocationRef {
+		return path.Builder{}.SplitUnescapeAppend(loc)
+	}
+
+	// We could get an empty LocationRef either because it wasn't populated or it
+	// was in the root of the data type.
+	elems := repoRef.Folders()
+
+	if ent.OneDrive != nil || ent.SharePoint != nil {
+		dp, err := path.ToDrivePath(repoRef)
+		if err != nil {
+			return nil, clues.Wrap(err, "fallback for LocationRef")
+		}
+
+		elems = append([]string{dp.Root}, dp.Folders...)
+	}
+
+	return path.Builder{}.Append(elems...), nil
+}
+
+func basicPathMerge(repoRef path.Path, locRef *path.Builder) (path.Path, error) {
+	if len(locRef.Elements()) == 0 {
+		res, err := path.ServicePrefix(
+			repoRef.Tenant(),
+			repoRef.ResourceOwner(),
+			repoRef.Service(),
+			repoRef.Category())
+		if err != nil {
+			return nil, clues.Wrap(err, "getting prefix for empty location")
+		}
+
+		return res, nil
+	}
+
+	return locRef.ToDataLayerPath(
+		repoRef.Tenant(),
+		repoRef.ResourceOwner(),
+		repoRef.Service(),
+		repoRef.Category(),
+		false)
+}
+
+func drivePathMerge(
+	ent *details.Entry,
+	repoRef path.Path,
+	locRef *path.Builder,
+) (path.Path, error) {
+	// Try getting the drive ID from the item. Not all details versions had it
+	// though.
+	var driveID string
+
+	if ent.SharePoint != nil {
+		driveID = ent.SharePoint.DriveID
+	} else if ent.OneDrive != nil {
+		driveID = ent.OneDrive.DriveID
+	}
+
+	// Fallback to trying to get from RepoRef.
+	if len(driveID) == 0 {
+		odp, err := path.ToDrivePath(repoRef)
+		if err != nil {
+			return nil, clues.Wrap(err, "fallback getting DriveID")
+		}
+
+		driveID = odp.DriveID
+	}
+
+	return basicPathMerge(
+		repoRef,
+		path.FormatDriveFolders(driveID, locRef.Elements()...))
+}
+
+func restorePathsForItem(
+	ctx context.Context,
+	backupVersion int,
+	ent *details.Entry,
+) (path.RestorePaths, error) {
+	res := path.RestorePaths{}
+
+	repoRef, err := path.FromDataLayerPath(ent.RepoRef, true)
+	if err != nil {
+		err = clues.Wrap(err, "parsing details path after reduction").
+			WithClues(ctx).
+			With("storage_path", clues.Hide(ent.RepoRef), "location", clues.Hide(ent.LocationRef))
+
+		return res, err
+	}
+
+	res.StoragePath = repoRef
+	ctx = clues.Add(ctx, "storage_path", repoRef)
```

Comment:
I like calling things what they are, whenever possible.
```suggestion
	ctx = clues.Add(ctx, "repo_ref", repoRef)
```